integration/tests/basic_functionality/negative/neg_deploy_with_invalid_r10k_yaml.rb

```diff
@@ -0,0 +1,42 @@
+require 'git_utils'
+require 'r10k_utils'
+require 'master_manipulator'
+test_name 'CODEMGMT-84 - C59271 - Attempt to Deploy with Invalid r10k Config'
+
+#Init
+env_path = on(master, puppet('config print environmentpath')).stdout.rstrip
+git_repo_path = '/git_repos'
+git_control_remote = File.join(git_repo_path, 'environments.git')
+
+r10k_config_path = '/etc/r10k.yaml'
+r10k_config_bak_path = "#{r10k_config_path}.bak"
+
+#In-line files
+r10k_conf = <<-CONF
+sources:
+  broken:
+    dir: "#{env_path}"
+    remote: "#{git_control_remote}"
+CONF
+
+#Verification
+error_message_regex = /ERROR.*can\'t\ convert\ nil\ into\ String/
```

Comment:
That is not a very informative error message, you should post a bug about making the error message more informative to the user.